### PR TITLE
Updated API serializers to support Namespaces and unskip unit tests

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #3748 
# What's Changed

- Updated API unit tests to no longer be skipped.
- Implemented API tests for Namespace
- Removed `namespace` from the `AvailableIPSerializer` and `AvailablePrefixSerializer` classes because it is inferred from the parent prefix
- [Documented some behavior/field changes for the IPAM API/models](https://github.com/nautobot/nautobot/pull/3869/commits/7835b3c5a7030fe3adf46eeae9de8f5d011ad9ac)

# TODO

- [x] VRF tests
- [x ] Prefix tests
- [x ] Parallel prefix tests
- [x ] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)

### CODE REVIEW

File: nautobot/ipam/api/serializers.py [hunk 1]
```diff
@@ -235,8 +240,19 @@ class Meta:
...
+            "parent": {"required": False},
         }
 
+    def validate(self, data):
+        namespace = data.get("namespace", None)
+        parent = data.get("parent", None)
+
+        # Only assert namespace/parent on create.
+        if self.instance is None and not any([namespace, parent]):
+            raise ValidationError({"__all__": "One of parent or namespace must be provided"})
```
Comment (glennmatthews) [inline nautobot/ipam/api/serializers.py:252]: This should be documented in the "API Behavior Changes" section of `upgrading-from-nautobot-v1.md`.

Comment (glennmatthews) [reply nautobot/ipam/api/serializers.py:252]: In fact I'm a bit concerned that the term `namespace` doesn't appear even once in that document presently... 😭

Comment (jathanism) [reply nautobot/ipam/api/serializers.py:252]: Now is the time!

File: nautobot/docs/installation/tables/v2-api-behavior-changes.yaml [hunk 2]
```diff
@@ -23,6 +23,12 @@
 - Model: "Interface"
   Field: "status"
   Changes: "Now is a foreign-key rather than a pseudo-enum"
+- Model: "IPAddress"
+  Field: "namespace"
+  Changes: "A derived field required on creation if `parent` isn't provided to find a correct parent Prefix"
+- Model: "IPAddress"
+  Field: "parent"
+  Changes: "A new foreign-key to `Prefix` required on creation if `namespace` isn't provided to find a correct parent Prefix"
```
[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/docs/installation/tables/v2-api-behavior-changes.yaml:31]: Also, should `parent` be documented in v2-database-behavior-changes.yaml as well?

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**